src/main/java/fr/jmmc/oiexplorer/core/gui/chart/RulerOverlay.java

```diff
@@ -0,0 +1,262 @@
+            g2.drawLine((int) originCoords.getX(), (int) originCoords.getY(), (int) endCoords.getX(), (int) endCoords.getY());
+
+            g2.setClip(chartPanel.getChartRenderingInfo().getChartArea());
+            g2.setColor(Color.black);
+            g2.drawString("Point 1: x=" + String.format("%.5f", origin.getX()) + " y=" + String.format("%.5f", origin.getY()), 100, chartPanel.getSize().height - 120);
+            g2.drawString("Point 2: x=" + String.format("%.5f", end.getX()) + " y=" + String.format("%.5f", end.getY()), 100, chartPanel.getSize().height - 105);
+            g2.drawString("Measure: " + String.format("%.5f", calculateMeasure()) + " mas", 100, chartPanel.getSize().height - 60);
+            g2.drawString("Angle: " + String.format("%.5f", calculateAngle()) + " °", 100, chartPanel.getSize().height - 45);
+        }
+    }
+
+    @Override
+    public boolean support(final int eventType) {
+        return true;
+    }
+
+    @Override
+    public void chartMouseMoved(final ChartMouseEvent chartMouseEvent) {
+        if (rulerState == RulerState.EDITING) {
+            end = translate(new PointDouble(
+                    chartMouseEvent.getTrigger().getX(),
+                    chartMouseEvent.getTrigger().getY()),
+                    Translation.COORDS_TO_VALUES
+            );
+            chartPanel.repaint();
+        }
+    }
+
+    @Override
+    public void chartMouseClicked(final ChartMouseEvent chartMouseEvent) {
+
+        switch (rulerState) {
+            case DISABLED:
+                rulerState = RulerState.DONE;
+            case DONE:
+                origin = translate(new PointDouble(
+                        chartMouseEvent.getTrigger().getX(),
+                        chartMouseEvent.getTrigger().getY()),
+                        Translation.COORDS_TO_VALUES
+                );
+                end = origin;
+                rulerState = RulerState.EDITING;
+                chartPanel.repaint();
+                break;
+
+            case EDITING:
+                end = translate(new PointDouble(
+                        chartMouseEvent.getTrigger().getX(),
+                        chartMouseEvent.getTrigger().getY()),
+                        Translation.COORDS_TO_VALUES
+                );
+                rulerState = RulerState.DONE;
+                chartPanel.repaint();
+                break;
+        }
+    }
+
+    /**
+     * MouseMotionListener for dragging
+     */
+    private class MotionListener implements MouseMotionListener {
+
+        private boolean doPointsIntersect(PointDouble p1, PointDouble p2, int spacing) {
+            return (p1.getX() < p2.getX() + spacing
+                    && p1.getX() > p2.getX() - spacing
+                    && p1.getY() < p2.getX() + spacing
+                    && p1.getY() > p2.getX() - spacing);
+        }
+
+        @Override
+        public void mouseDragged(MouseEvent e) {
+            PointDouble p = translate(new PointDouble(
```

Comment:
factorize the new PointDouble( e.getX(),e.getY())

Reply:
You should ensure the moved point remains in visible area (plot box) like bound checks.

Reply:
The points are only visible in the plot but can be outside it without being visible. Thus, we can zoom in on the image and the ruler goes outside the plot without being visible.
![image](https://user-images.githubusercontent.com/59568659/124615556-ae323d00-de75-11eb-81c4-402e1a982d53.png)